implementation/src/main/java/io/smallrye/health/HealthStatus.java

```diff
@@ -64,4 +65,14 @@ public static HealthCheck state(String name, boolean state) {
         return () -> HealthCheckResponse.named(name).state(state).build();
     }
 
+    private static synchronized final String generateRandomHealthCheckName() {
+        // lazy init to avoid loading a random object when not necessary
+        if (random == null) {
+            random = new Random();
+        }
+
+        int suffix = random.nextInt(999);
+        return String.format("random-health-check-%03d", suffix);
```

Comment:
I would only suggest a different name, something like maybe "status-health-check-XYZ" or "generated-health-check-XYZ". "random-" may mean more things in this context.

Reply:
True in fact I had `generated` in mind but when I was going to write I had an interruption, and then I put my original idea of `random`

Reply:
Agree, I'd probably call it something like `unnamed-1234567890` :-)